### PR TITLE
updated docstring for pair_programming

### LINKED ISSUE #825: pair_programming for git backend doesn't work right
So I was exploring the Git enricher and came through this issue with  `pair_programming` set to `true` in `setup_cfg`.

The variables `m` and `n` set [here](https://github.com/chaoss/grimoirelab-elk/blob/8ea3455bd639de695e1a4d2f56ed86df4e5818c7/grimoire_elk/enriched/git.py#L414) through regex don't seem to work for the repository with multi-authored commits I've chosen (https://github.com/juliettepretot/npx-co-commit).

Based on the current logic both the variables are set to `None` because the regex used in `AUTHOR_P2P_REGEX` and `AUTHOR_P2P_NEW_REGEX` seems faulty, when they both should get populated by the 2 author names for that particular commit.

For instance let's look at this commit JSON data from Perceval 
```
{
    "backend_name": "Git",
    "backend_version": "0.12.0",
    "category": "commit",
    "classified_fields_filtered": null,
    "data": {
        "Author": "Juliette Pretot <hi@juliette.sh>",
        "AuthorDate": "Mon Oct 29 16:29:48 2018 +0100",
        "Commit": "Juliette Pretot <hi@juliette.sh>",
        "CommitDate": "Tue Nov 6 21:41:35 2018 +0100",
        "commit": "dbc4e2c18c893be4db75b472d4016e623316729e",
        "files": [
            {
                "action": "A",
                "added": "10",
                "file": ".gitignore",
                "indexes": [
                    "0000000",
                    "e45672e"
                ],
                "modes": [
                    "000000",
                    "100644"
                ],
                "removed": "0"
            },
            {
                "action": "A",
                "added": "24",
                "file": "README.MD",
                "indexes": [
                    "0000000",
                    "45d1af5"
                ],
                "modes": [
                    "000000",
                    "100644"
                ],
                "removed": "0"
            },
            {
                "action": "A",
                "added": "46",
                "file": "index.js",
                "indexes": [
                    "0000000",
                    "5f2990c"
                ],
                "modes": [
                    "000000",
                    "100755"
                ],
                "removed": "0"
            },
            {
                "action": "A",
                "added": "27",
                "file": "package-lock.json",
                "indexes": [
                    "0000000",
                    "d0ef368"
                ],
                "modes": [
                    "000000",
                    "100644"
                ],
                "removed": "0"
            },
            {
                "action": "A",
                "added": "19",
                "file": "package.json",
                "indexes": [
                    "0000000",
                    "8f521a3"
                ],
                "modes": [
                    "000000",
                    "100644"
                ],
                "removed": "0"
            },
            {
                "action": "A",
                "added": "-",
                "file": "screenshot.gif",
                "indexes": [
                    "0000000",
                    "74de826"
                ],
                "modes": [
                    "000000",
                    "100644"
                ],
                "removed": "-"
            }
        ],
        "message": "initial commit\n\nCo-authored-by: mariiapunda <mariiapunda@users.noreply.github.com>",
        "parents": [],
        "refs": []
    },
    "origin": "https://github.com/juliettepretot/npx-co-commit",
    "perceval_version": "0.12.24",
    "search_fields": {
        "item_id": "dbc4e2c18c893be4db75b472d4016e623316729e"
    },
    "tag": "https://github.com/juliettepretot/npx-co-commit",
    "timestamp": 1585582796.576738,
    "updated_on": 1541536895.0,
    "uuid": "0c9876869058dc5b3f836365f3ce06a5b5575749"
}
```
According to the above commit data there must be 2 authors for the commit **Juliette Pretot** and **mariiapunda**  (based on the line `"message": "initial commit\n\nCo-authored-by: mariiapunda `)

But after fixing regex 
```
AUTHOR_P2P_REGEX = re.compile(r'(?P<first_authors>.* .*) and (?P<last_author>.* .*) (?P<email>.*)')
```
 to
```
AUTHOR_P2P_REGEX = re.compile(r'(?P<first_authors>.* ) (?P<last_author> .*) (?P<email>.*)')
```
what I observe is in the function `__get_authors()` [here](https://github.com/chaoss/grimoirelab-elk/blob/8ea3455bd639de695e1a4d2f56ed86df4e5818c7/grimoire_elk/enriched/git.py#L118) the committer name from this line 

```
"Author": "Juliette Pretot <hi@juliette.sh>",
```

is used and gets split into two names and saved in `item['data']['authors']` as `<class 'list'>: ['Juliette', 'Pretot']`.

AFAIK `item['data']['authors']` in this case should be `<class 'list'>: ['Juliette Pretot', 'mariiapunda']`.

I feel the logic being used here is wrong, from the regex to how multiple authors are being identified.
Please tell me if I am wrong with the repository I chose or the way I am interpreting the logic.

### PR DESCRIPTION
**Pull request by imnitishng**
This PR adds docstring for `pair_programming` enabled mode in Git backend based on the discussions [here](https://github.com/chaoss/grimoirelab-elk/issues/825)
Fixes #825 

Signed-off-by: Nitish Gupta <imnitish.ng@gmail.com>

### CODE REVIEW

File: grimoire_elk/enriched/git.py [hunk 1]
```diff
@@ -388,7 +388,34 @@ def get_pair_programming_metrics(eitem, nauthors):
         return eitem
 
     def enrich_items(self, ocean_backend, events=False):
-        """ Implementation supporting signed-off and multiauthor/committer commits."""
+        """ Implementation supporting signed-off and multiauthor/committer commits.
+            Multiauthor/Multcommiter commits are the ones authored/commited by more
```
Comment (valeriocos) [inline grimoire_elk/enriched/git.py:392]: Please align all the text at the same level of the first `"""` (like the example below).
```
""" Implementation supporting signed-off and multiauthor/committer commits.
Multiauthor/Multcommiter commits are the ones authored/commited by more
...
```

File: grimoire_elk/enriched/git.py [hunk 2]
```diff
@@ -388,7 +388,34 @@ def get_pair_programming_metrics(eitem, nauthors):
...
+                "CommitDate": "Tue Aug 14 14:32:15 2012 -0300",
+                "Signed-off-by": [
+                    "Eduardo Morais <companheiro.vermelho@gmail.com>"
+                ],
+                "commit": "87783129c3f00d2c81a3a8e585eb86a47e39891a",
+            ...
+
+            authors extracted are ['Eduardo Morais', 'Zhongpeng Lin'] (can be more than 2).
+            A new rich item is now created using each author name. For multicommitter only
+            the first committer name is used to create the rich item. Incase the commit is
```
Comment (valeriocos) [inline grimoire_elk/enriched/git.py:410]: typo `Incase`

File: grimoire_elk/enriched/git.py [hunk 3]
```diff
@@ -388,7 +388,34 @@ def get_pair_programming_metrics(eitem, nauthors):
...
+            signed-off by committer, raw data has extra "Signed-off-by" attribute used to
+            create a new rich item for every author who signed off the commit, making sure
+            duplicate entries are not created.
+
+            Note -
+                "message": "Enable users to pass flags\n\nCo-authored-by: mariiapunda <mariiapunda@users.noreply.github.com>",
+            Co-authored commits like these are not considered as multiauthored commits in ELK.
+            """
```
Comment (valeriocos) [inline grimoire_elk/enriched/git.py:418]: Please remove the empty line after the docstring.

**Review by valeriocos — COMMENTED** [review]
Thank you @imnitishng for the PR! This doc will ease the maintenance of the pair programming logic. I left some minor comments. Thanks for your work!

**Review by valeriocos — APPROVED** [review]
Please squash the commits into one, and then I guess we are ready to go :)

**Comment by imnitishng** [comment]
Sorry I did something wrong, figuring it out please wait.

**Comment by imnitishng** [comment]
Done.

**Comment by valeriocos** [comment]
Thanks @imnitishng !

**Pull request merged**